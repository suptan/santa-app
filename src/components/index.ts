export * from "./Button";
export * from "./Form";
export * from "./Footer";
export * from "./Header";
export * from "./Input";
export * from "./Loader";
export * from "./Wish";
