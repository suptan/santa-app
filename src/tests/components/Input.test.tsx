import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Input } from "@/components";

describe("<Input>", () => {
  it.each`
    type
    ${"input"}
    ${"textarea"}
  `("should render as $type", ({ type }) => {
    const { container } = render(<Input type={type} />);
    expect(container).toMatchSnapshot();
  });
});
