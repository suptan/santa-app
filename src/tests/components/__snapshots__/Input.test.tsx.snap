// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> should render as input 1`] = `
<div>
  <div
    class="flex flex-col gap-1 text-gray-400"
  >
    <div
      class="border border-zinc-200 rounded bg-transparent"
    >
      <input
        class="bg-transparent p-2 w-full"
        type="input"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<Input> should render as textarea 1`] = `
<div>
  <div
    class="flex flex-col gap-1 text-gray-400"
  >
    <div
      class="border border-zinc-200 rounded bg-transparent"
    >
      <textarea
        class="bg-transparent p-2 w-full"
        type="textarea"
      />
    </div>
  </div>
</div>
`;
